OpenSim/Common/Component.h

```diff
@@ -813,7 +811,19 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
         auto it = _connectorsTable.find(name);
 
         if (it != _connectorsTable.end()) {
-            return get_connectors(it->second);
+            // The following allows one to use a Connector immediately after
+            // copying the component;
+            // e.g., myComponent.clone().getConnector("a").getConnecteeName().
+            // Since we use the default copy constructor for Component,
+            // the copied AbstractConnector cannot know its new owner
+            // immediately after copying.
+            if (!it->second->hasOwner()) {
+                // The `this` pointer must be non-const because the Connector
+                // will want to be able to modify the connectee_name property.
+                const_cast<AbstractConnector*>(it->second.get())->setOwner(
+                        const_cast<Self&>(*this));
```

Comment:
@aymanhab I have added your comment to #1393. For this PR, would it be sufficient to add a note to the function doxygen block saying that there is such a side effect?